src/components/pinning-manager/PinningManager.js

```diff
@@ -0,0 +1,151 @@
+import GlyphPin from '../../icons/GlyphPin'
+import ContextMenu from '../context-menu/ContextMenu'
+import ContextMenuItem from '../context-menu/ContextMenuItem'
+import GlyphDots from '../../icons/GlyphDots'
+import StrokeSettings from '../../icons/StrokeSettings'
+import StrokeCancel from '../../icons/StrokeCancel'
+import StrokeExternalLink from '../../icons/StrokeExternalLink'
+
+import './PinningManager.css'
+
+const ROW_HEIGHT = 50
+const HEADER_HEIGHT = 32
+
+export const PinningManager = ({ pinningServices, doFilesSizeGet, doFilesFetch, filesSize, t }) => {
+  const [isModalOpen, setModalOpen] = useState(false)
+  // const onModalOpen = () => setModalOpen(true)
+  const onModalClose = () => setModalOpen(false)
+
+  const [sortSettings, setSortSettings] = useState({
+    sortBy: 'addedAt',
+    sortDirection: SortDirection.ASC
+  })
+  useEffect(() => {
+    (async () => {
+      await doFilesFetch()
+      await doFilesSizeGet()
+    })()
+  }, [doFilesFetch, doFilesSizeGet])
+
+  const localPinning = useMemo(() =>
+    ({ name: t('localPinning'), type: 'LOCAL', svgIcon: GlyphPin, totalSize: filesSize }),
+  [filesSize, t])
+
+  const sortedServices = useMemo(() =>
+    (pinningServices || []).sort(sortByProperty(sortSettings.sortBy, sortSettings.sortDirection === SortDirection.ASC ? 1 : -1)),
+  [pinningServices, sortSettings.sortBy, sortSettings.sortDirection])
+
+  const sortedList = useMemo(() => [localPinning, ...sortedServices], [localPinning, sortedServices])
+
+  return (
+    <Fragment>
+      <div className="mv4 pinningManager">
+        <div className='ph1 ph3-l flex items-center bg-white lh-copy charcoal f6 fw5'>
+          <AutoSizer disableHeight>
+            {({ width }) => (
+              <Table
+                className='tl fw4 w-100 f6'
+                headerClassName='gray ttc fw4 f6 ph2'
+                width={width}
+                height={(sortedList.length + 1) * ROW_HEIGHT}
+                headerHeight={HEADER_HEIGHT}
+                rowHeight={ROW_HEIGHT}
+                rowCount={sortedList.length}
+                rowGetter={({ index }) => sortedList[index]}
+                rowClassName='bb b--light-gray mt2'
+                sort={(...sortArgs) => setSortSettings(...sortArgs)}
+                sortBy={sortSettings.sortBy}
+                sortDirection={sortSettings.sortDirection}>
+                <Column label={t('service')} title={t('service')} dataKey='name' width={width * 0.4} flexShrink={0} flexGrow={1} cellRenderer={ServiceCell} className='charcoal truncate f6' />
+                <Column label={t('files')} title={t('files')} dataKey='totalSize' width={width * 0.2} flexShrink={0} cellRenderer={SizeCell} className='charcoal truncate f6 pl2' />
+                <Column label={t('bandwidthUsed')} title={t('bandwidthUsed')} dataKey='bandwidth' width={width * 0.2} flexShrink={1} cellRenderer={BandwidthCell} className='charcoal truncate f6 pl2' />
```

Comment:
@lidel Good memory - thank you for calling that out!
Agree that indicating the number of items pinned is the best choice, particularly since we're removing that from the info bar in the Files screen. However, I'd suggest just `Pins` for the column title so it doesn't truncate super-easily (IMO the fact that there is no unit in the cells for the Pins column, as opposed to "MB" units for the Files column, implies quantity without any need for further explanation).

Reply:
Done ✅